repository.go

```diff
@@ -586,7 +586,8 @@ func (r repository) saveHasOne(cw contextWrapper, doc *Document, mutation *Mutat
 			assocMut         = assocMuts.Mutations[0]
 		)
 
-		if loaded {
+		if loaded && ("" == assoc.ForeignField() ||
+			("" != assoc.ForeignField() && !isZero(assoc.ForeignValue()))) {
```

Comment:
isn't this enough? `loaded && (assoc.ForeignField() == "" || !isZero(assoc.ForeignValue()))`

Reply:
@Fs02 you're right. I've fixed this. Many thanks for your review.